File: customview/src/main/java/com/dunzhixuan/app/customview/MainActivity.java
package com.dunzhixuan.app.customview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Animation shake = AnimationUtils.loadAnimation(MainActivity.this, R.anim.view_scroll);
                button.startAnimation(shake);
//                button.clearAnimation();
            }
        });
    }
}

File: reflex/src/main/java/com/dunzhixuan/reflex/MainActivity.kt
package com.dunzhixuan.reflex

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import java.lang.reflect.Field

class MainActivity : AppCompatActivity() {

    private var url: String = "https://mobile.vipkid.com.cn/parents/course/courseschedule/?_tbc=h"
    private val TAG : String = "MainActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.btn_reflex).setOnClickListener { getField() }
    }

    fun getField() {
        try {
            val clazz = Class.forName("com.dunzhixuan.reflex.MainActivity")
            var url : Field = clazz.getDeclaredField("url")
            Log.e(TAG, url.toString());
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
